editor/gui/editor_bottom_panel.cpp

```diff
@@ -59,6 +76,34 @@ void EditorBottomPanel::_switch_by_control(bool p_visible, Control *p_control) {
 	}
 }
 
+void EditorBottomPanel::_scroll(bool p_right) {
+	HScrollBar *h_scroll = button_scroll->get_h_scroll_bar();
+	if (Input::get_singleton()->is_key_pressed(Key::CTRL)) {
+		h_scroll->set_value(p_right ? h_scroll->get_max() : 0);
+	} else if (Input::get_singleton()->is_key_pressed(Key::SHIFT)) {
+		h_scroll->set_value(h_scroll->get_value() + (h_scroll->get_page() * (p_right ? 1 : -1)));
+	} else {
+		h_scroll->set_value(h_scroll->get_value() + ((h_scroll->get_page() * 0.5) * (p_right ? 1 : -1)));
+	}
+	_update_disabled_buttons();
+}
+
+void EditorBottomPanel::_update_scroll_buttons() {
+	bool show_arrows = button_hbox->get_size().width > button_scroll->get_size().width;
+	left_button->set_visible(show_arrows);
+	right_button->set_visible(show_arrows);
+
+	if (show_arrows) {
+		callable_mp(this, &EditorBottomPanel::_update_disabled_buttons).call_deferred();
```

Comment:
`_update_scroll_buttons()` is always deferred, so `_update_disabled_buttons()`, so this call doesn't need to (it doesn't even make difference in this case).